apollo-federation/src/operation/simplify.rs

```diff
@@ -311,16 +305,18 @@ impl InlineFragmentSelection {
                 // applied recursively. This could be worth investigating.
                 let rebased_inline_fragment =
                     self.inline_fragment.rebase_on(parent_type, schema)?;
-                let mut mutable_selections = self.selection_set.selections.clone();
```

Comment:
This was the cause: when lifting selections, we were looking at the un-flattened original selection set, instead of the flattened one.